alignak_module_backend/broker/module.py

```diff
@@ -437,6 +505,215 @@ def send_to_backend(self, type_data, name, data):
                 ret = False
         return ret
 
+    def update_status(self, brok):
+        # pylint: disable=too-many-locals
+        """We manage the status change for a backend host/service
+
+        :param brok: the brok
+        :type brok:
+        :return: None
+        """
+        if 'contact_name' in brok.data:
+            contact_name = brok.data['contact_name']
+            if brok.data['contact_name'] not in self.mapping['user']:
+                logger.warning("Got a brok for an unknown user: '%s'", contact_name)
+                return
+            endpoint = 'user'
+            name = contact_name
+            item_id = self.mapping['user'][name]
+        else:
+            host_name = brok.data['host_name']
+            if brok.data['host_name'] not in self.mapping['host']:
+                logger.warning("Got a brok for an unknown host: '%s'", host_name)
+                return
+            endpoint = 'host'
+            name = host_name
+            item_id = self.mapping['host'][name]
+            if 'service_description' in brok.data:
+                service_name = '__'.join([host_name, brok.data['service_description']])
+                endpoint = 'service'
+                name = service_name
+                item_id = self.mapping['service'][name]
+                if service_name not in self.mapping['service']:
+                    logger.warning("Got a brok for an unknown service: '%s'", service_name)
+                    return
+
+        # Sort brok properties
+        sorted_brok_properties = sorted(brok.data)
+        logger.debug("Update status %s: %s", endpoint, sorted(brok.data))
+
+        # Search the concerned element
+        item = self.backend.get(endpoint + '/' + item_id)
```

Comment:
For perfs, I prefer get all items of an endpoint at init of the module instead make many time same backend query